jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -233,6 +325,66 @@ static JarType determineJarType(Path jarPath) throws IOException {
     }
   }
 
+  /**
+   * Creates layers as specified by the layers.idx file (located in the BOOT-INF/ directory of the
+   * JAR).
+   *
+   * @param localExplodedJarRoot Path to temporary directory
+   * @return list of {@link FileEntriesLayer}
+   * @throws IOException if temporary directory provided doesn't exist
+   */
+  private static List<FileEntriesLayer> createLayersForLayeredSpringBootJar(
+      Path localExplodedJarRoot) throws IOException {
+    Path layerIndexPath = localExplodedJarRoot.resolve(Paths.get("BOOT-INF/layers.idx"));
+    Pattern layerNamePattern = Pattern.compile("-\\s(.*):");
+    Pattern fileNamePattern = Pattern.compile("\\s\\s-\\s(.*)");
```

Comment:
The [Spring doc](https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-executable-jar-format.html#executable-jar-war-index-files-layers) makes it very explicit that they are either dash-space or space-space-dash-space. I think we can use ` ` (space) instead of `\\s`. Also the doc says paths and layer names are quoted. So we can have it like `Pattern.compile("- \"([^\"]*)\":")` (untested) without later removing quotes?